File: scala/Euler6.scala
/*
Find the difference between the sum of the squares of the first one hundred natural numbers 
and the square of the sum.*/
object Euler6 extends App{
	val squareOfSumOfN: Long => Long = n => square(((n*(n+1)) / 2))
	val sumOfSquareOfN: Long => Long = n =>  ((n * (n+1) * ((2*n)+1)) / 6)
	val square: Long => Long = (x) => x * x
  
	def solution(num: Long):Long = {
	  squareOfSumOfN(num) - sumOfSquareOfN(num) 
	}
	
	println(solution(100))
	
}

File: python/Euler6.py
"""
Find the difference between the sum of the squares of the first one hundred natural numbers
and the square of the sum.
"""

square = lambda n: n * n
squareOfSumOfN = lambda n: square((n*(n+1))/2)
sumOfSquareOfN = lambda n: ((n * (n+1) * ((2*n)+1)) / 6)

def solution(num):
    return squareOfSumOfN(num) - sumOfSquareOfN(num)
                
